### PR TITLE
test(clusterconfig): add unit test for recent_metrics.go and rename it to gather_most_recent_metrics.go

### PR DESCRIPTION
**Pull request by rluders**
This PR adds a simple unit test for the `most recent metrics` gatherer and renames its file from `recent_metrics.go -> gather_most_recent_metrics.go`

## Categories

- [ ] Bugfix
- [ ] Data Enhancement
- [ ] Feature
- [ ] Backporting
- [X] Others (CI, Infrastructure, Documentation)

## Sample Archive

None

## Documentation

None

## Unit Tests

- `pkg/gatherers/clusterconfig/gather_most_recent_metrics_test.go`

## Privacy

Yes. There are no sensitive data in the newly collected information.

## Breaking Changes

No

## References

https://issues.redhat.com/browse/CCXDEV-10120

### CODE REVIEW

File: pkg/gatherers/clusterconfig/gather_most_recent_metrics_test.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+				}},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			records, errs := gatherMostRecentMetrics(ctx, tt.metricsClient.RestClient(t))
+			assert.Equalf(t, tt.wantRecords, records, "gatherMostRecentMetrics(%v, %v)", ctx, tt.metricsClient)
+			assert.Equalf(t, tt.wantErrors, errs, "gatherMostRecentMetrics(%v, %v)", ctx, tt.metricsClient)
```
Comment (tremes) [inline pkg/gatherers/clusterconfig/gather_most_recent_metrics_test.go:92]: The message `gatherMostRecentMetrics(%v, %v)", ctx, tt.metricsClient` doesn't seem very useful TBH.

**Comment by rluders** [comment]
Just a unit test update, e2e can be ignored.
/override ci/prow/e2e-agnostic-upgrade ci/prow/e2e-metal-ipi-ovn-ipv6 ci/prow/insights-operator-e2e-tests

**Comment by tremes** [comment]
A new unit test. Reviewed.
/lgtm

**Pull request merged**